src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java

```diff
@@ -331,39 +331,58 @@ private void buildSscsDocumentFromScan(SscsCaseData sscsCaseData, State caseStat
             preSubmitCallbackResponse.addError("No further evidence to process");
         }
 
+        addedDocumentsUtil.computeDocumentsAddedThisEvent(sscsCaseData, documentsAddedThisEvent, EVENT_TYPE);
         sscsCaseData.setScannedDocuments(null);
     }
 
-    private void buildSscsDocuments(SscsCaseData sscsCaseData, ScannedDocument scannedDocument, State caseState) {
+    private void addSscsDocumentToCaseData(SscsCaseData sscsCaseData, SscsDocument sscsDocument) {
         List<SscsDocument> documents = new ArrayList<>();
+        documents.add(sscsDocument);
 
-        if (!equalsIgnoreCase(scannedDocument.getValue().getType(), COVERSHEET)) {
-
-            SscsDocument sscsDocument = buildSscsDocument(sscsCaseData, scannedDocument, caseState);
+        if (sscsCaseData.getSscsDocument() != null) {
+            documents.addAll(sscsCaseData.getSscsDocument());
+        }
 
-            if (REINSTATEMENT_REQUEST.getValue().equals(sscsDocument.getValue().getDocumentType())) {
-                if (isFurtherEvidenceActionCode(sscsCaseData.getFurtherEvidenceAction(), OTHER_DOCUMENT_MANUAL.getCode())) {
-                    setReinstateCaseFields(sscsCaseData);
-                }
-            }
+        sscsCaseData.setSscsDocument(documents);
```

Comment:
Didn't this used to get executed for coversheets as well?

Reply:
Do you mean the documents being set? The coversheet was never added to the list of documents before: https://github.com/hmcts/sscs-tribunals-case-api/blob/master/src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/furtherevidence/actionfurtherevidence/ActionFurtherEvidenceAboutToSubmitHandler.java#L338-L366